worker_test.go

```diff
@@ -0,0 +1,223 @@
+package main
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+type testfile struct {
+	path   string
+	data   []byte
+	upload string
+}
+
+func tempDirFromTestFiles(t *testing.T, testfiles []testfile) string {
+	// generate temp directory and write out all test files
+	root := filepath.Join(t.TempDir(), "data")
+
+	for _, file := range testfiles {
+		path := filepath.Join(root, file.path)
+
+		// ensure parents exist
+		if err := os.MkdirAll(filepath.Dir(path), 0o755); err != nil {
+			t.Fatal(err)
+		}
+
+		// write file
+		if err := os.WriteFile(path, file.data, 0o644); err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	return root
+}
+
+type pair struct{ src, dst string }
+
+type mockUploader struct {
+	Error   error
+	Uploads map[pair]bool
+}
+
+func newMockUploader() *mockUploader {
+	return &mockUploader{
+		Uploads: make(map[pair]bool),
+	}
+}
+
+func (up *mockUploader) UploadFile(src, dst string, meta *MetaData) error {
+	if up.Error != nil {
+		return up.Error
+	}
+	up.Uploads[pair{src, dst}] = true
+	return nil
+}
+
+func scanAndProcessDirOnce(worker *Worker, root string) error {
+	stop := make(chan struct{})
+	defer close(stop)
+	jobs := make(chan Job)
+
+	go func() {
+		scanForJobs(stop, jobs, root)
+		close(jobs)
+	}()
+
+	for job := range jobs {
+		worker.Process(job)
+	}
+
+	return nil
+}
+
+func TestScanAndProcess(t *testing.T) {
+	testcases := map[string][]testfile{
+		"NoJob": {
```

Comment:
Do you have a test for the wrong path?

Reply:
Or a file that would not be caught by your file parser in `scanForJobs`?  I am curious how the code behaves when there are "lost files"

Reply:
Good catch. I added some additional tests for this which actually caught and a case I missed.

I'm using stricter patterns now to address those failing tests.